src/relation/mod.rs

```diff
@@ -935,36 +946,49 @@ impl Join {
             .into_iter()
             .zip(left_schema.iter())
             .map(|(name, field)| {
-                Field::new(
-                    name,
-                    if transform_datatype_in_optional_left {
-                        DataType::optional(field.data_type())
-                    } else {
-                        field.data_type()
-                    },
-                    if right_is_unique {
-                        field.constraint()
-                    } else {
+                let (data_type, constraint) = if (
```

Comment:
Why not match the operator ? or the pair (operator, right_schema.field(&field.name())) ?